File: ympa_fsharp_giraffe_server/src/model/GetPromoDTO.fs
namespace ympa_fsharp_giraffe_server.Model

open System
open System.Collections.Generic
open ympa_fsharp_giraffe_server.Model.ChannelType
open ympa_fsharp_giraffe_server.Model.GetPromoAssortmentInfoDTO
open ympa_fsharp_giraffe_server.Model.GetPromoBestsellerInfoDTO
open ympa_fsharp_giraffe_server.Model.GetPromoConstraintsDTO
open ympa_fsharp_giraffe_server.Model.GetPromoMechanicsInfoDTO
open ympa_fsharp_giraffe_server.Model.PromoPeriodDTO

module GetPromoDTO =

  //#region GetPromoDTO


  type GetPromoDTO = {
    Id : string;
    Name : string;
    Period : PromoPeriodDTO;
    Participating : bool;
    AssortmentInfo : GetPromoAssortmentInfoDTO;
    MechanicsInfo : GetPromoMechanicsInfoDTO;
    BestsellerInfo : GetPromoBestsellerInfoDTO;
    Channels : ChannelType[];
    Constraints : GetPromoConstraintsDTO;
  }
  //#endregion

File: src/main/kotlin/ympa_kotlin_wiremock_server/models/GetPromoDTO.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

data class GetPromoDTO(
    @field:JsonProperty("id")
    val id: kotlin.String,

    @field:JsonProperty("name")
    val name: kotlin.String,

    @field:JsonProperty("period")
    val period: PromoPeriodDTO,

    @field:JsonProperty("participating")
    val participating: kotlin.Boolean,

    @field:JsonProperty("assortmentInfo")
    val assortmentInfo: GetPromoAssortmentInfoDTO,

    @field:JsonProperty("mechanicsInfo")
    val mechanicsInfo: GetPromoMechanicsInfoDTO,

    @field:JsonProperty("bestsellerInfo")
    val bestsellerInfo: GetPromoBestsellerInfoDTO,

    @field:JsonProperty("channels")
    val channels: kotlin.collections.List<ChannelType>? = null,

    @field:JsonProperty("constraints")
    val constraints: GetPromoConstraintsDTO? = null,

)
